src/Traits/ReflectionClassLikeTrait.php

```diff
@@ -442,7 +443,7 @@ public function getParentClass()
             $extendsNode = $hasExtends ? $this->classLikeNode->$extendsField : null;
             if ($extendsNode instanceof FullyQualified) {
                 $extendsName = $extendsNode->toString();
-                $parentClass = class_exists($extendsName, false) ? new parent($extendsName) : new static($extendsName);
+                $parentClass = class_exists($extendsName, false) ? new BaseReflectionClass($extendsName) : new ReflectionClass($extendsName);
```

Comment:
Hmm... Here I'm a bit torn here:

- I don't like the old code: Scrutinize is right.
  + This is a trait. in theory, it should be includable in any class
  + Traits don't have parents, or even self.
    * PHP did the right thing in interpreting `self` and `parent` relative to the trait user
  + This interpretation of `self` and `parent` is rather non-obvious, and does nothing to aid code clarity.
  + There is no way to statically analyze this without the class it's being used in, which explains Scrutinize's issue.
- I don't like the new code either:
  + My changes totally lock this down to inclusion in one specific class: That's wrong too.
  + There really should be specified by the including class. a variable could work, but I think a class method in the using class is the best solution.
- One of the PRs I'm hoping to submit soon is a factory for reflection classes:
  + ... so I expect this issue to go away soon.

@aik099, for now, I think the best solution is a method in the using class to return the newly created objects. What do you think?